### PR TITLE
Fixing a bug in fieldset.from_netcdf() to check if two variables are on same grid

### LINKED ISSUE #546: Keyerror when calling computeTimeChunk
Hi! after updating the master branch yesterday, I got a KeyError I didn't have before... It seems somehow it's not reading the variables correctly. I add three fields of data, a 3D velocity field  (fieldset.from_nemo), and temperature and salinity from a regular field (field.from_netcdf).

Traceback (most recent call last):
  File "/home/valles/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/xarray/core/dataset.py", line 887, in _construct_dataarray
    variable = self._variables[name]
KeyError: 'vomecrty'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "run_NBC_GLORYS2V4_TRY.py", line 147, in <module>
    T=fieldset.computeTimeChunk(fieldset.U.grid.time_full[0], 1)
  File "/home/valles/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/parcels-2.0.0b1.dev35+g165e476-py3.7.egg/parcels/fieldset.py", line 591, in computeTimeChunk
    f.computeTimeChunk(data, tind)
  File "/home/valles/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/parcels-2.0.0b1.dev35+g165e476-py3.7.egg/parcels/field.py", line 927, in computeTimeChunk
    if len(filebuffer.data.shape) == 2:
  File "/home/valles/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/parcels-2.0.0b1.dev35+g165e476-py3.7.egg/parcels/field.py", line 1438, in data
    data = self.dataset[self.name]
  File "/home/valles/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/xarray/core/dataset.py", line 961, in __getitem__
    return self._construct_dataarray(key)
  File "/home/valles/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/xarray/core/dataset.py", line 890, in _construct_dataarray
    self._variables, name, self._level_coords, self.dims)
  File "/home/valles/miniconda3/envs/py3_parcels/lib/python3.7/site-packages/xarray/core/dataset.py", line 76, in _get_virtual_variable
    ref_var = variables[ref_name]
KeyError: 'vomecrty'

### PR DESCRIPTION
**Pull request by erikvansebille**
Now also checking if filenames for the two variables are the same, and testing using nemo_3d data. This fixes #546

### CODE REVIEW

File: parcels/examples/example_nemo_curvilinear.py [hunk 1]
```diff
@@ -68,6 +69,29 @@ def test_nemo_curvilinear(mode):
     run_nemo_curvilinear(mode, outfile)
 
 
+def test_nemo_3D_samegrid():
+    data_path = path.join(path.dirname(__file__), 'NemoNorthSeaORCA025-N006_data/')
```
Comment (delandmeterp) [inline parcels/examples/example_nemo_curvilinear.py]: So, you're adding new data in oceanparcels_website?

File: parcels/fieldset.py [hunk 2]
```diff
@@ -253,7 +252,9 @@ def from_netcdf(cls, filenames, variables, dimensions, indices=None,
             for procvar, _ in fields.items():
                 procdims = dimensions[procvar] if procvar in dimensions else dimensions
                 procinds = indices[procvar] if (indices and procvar in indices) else indices
-                if procdims == dims and procinds == inds:
+                procpaths = filenames[procvar] if type(filenames) is dict and procvar in filenames else filenames
```
Comment (delandmeterp) [inline parcels/fieldset.py]: I try to use `isinstance(filenames, dict)`
except if I check for two types, then I use `type(var) in [type1, type2]`

Do you know if there is a preferred option between isinstance and type ?

File: parcels/examples/example_nemo_curvilinear.py (hunk 1 above)
Comment (erikvansebille) [reply parcels/examples/example_nemo_curvilinear.py]: No, this is a new unit test that uses the 3D nemo data that was there already for the 3d nemo tutorial. The point is that we want to test this on multiple filenames

Comment (delandmeterp) [reply parcels/examples/example_nemo_curvilinear.py]: Oh yes sure. I was thinking there are only two time steps, so no defer loading, but it is not relevant in this test. my bad

**Pull request merged**